File: react-type-vite/src/services/api/response/courseResponse.ts
import type { EducationalUnitResponse } from "./educationalUnitResponse";
import type { TeacherResponse } from "./teacherResponse";

export interface CourseResponse {
  id: number;
  courseName: string;
  description?: string;
  credits?: number;
  maxStudents?: number;
  currentStudents?: number;
  createdAt?: Date;
  updatedAt?: Date;
  idTeacher?: string;
  teacher?: TeacherResponse;
  institution?: EducationalUnitResponse;
  sections?: any[];
}

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/CourseResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CourseResponse {
    private Integer id;

    private String courseName;

    private String description;

    private Integer credits;

    private Integer maxStudents;

    private Integer currentStudents;

    private Date createdAt;

    private Date updatedAt;

    private String idTeacher;

    private TeacherResponse teacher;

    private Integer idEducationalUnit;
}
